### DISCUSSION TITLE
Can't use NetDaemon.Extensions.Logging in a Container because of Roslynator.Analyzers & native?

### DISCUSSION THREAD

**Discussion by ab5000** [net-daemon/netdaemon#1100] [General]
Hi all, (apologies below*)

I was trying to package a Console app I was playing with as a Docker container, but was getting a weird error about the .NET framework not being found.  Even though the docker built and ran fine as a "Hello World" console.  (I think I've gone a little rogue, because I am using the netdaemon nuget packages for my app)

I removed the netdaemon packages one-by-one until I isolated the NetDaemon.Extensions.Logging package as the one that was breaking my container.  When I look in the csproj, I noticed this:
<PackageReference Include="Roslynator.Analyzers" Version="4.12.2">
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
      <PrivateAssets>all</PrivateAssets>
    </PackageReference>
    
I'm guessing that the native part is a Windows only thing.

But like I said, I've gone rogue, so I might be totally off-base.

* apologies if I'm off-base or doing this wrong.  I'm new to this, and learning more every day...  I didn't report this as an issue, because I'm not sure what I'm doing is a supported pattern. 

Thanks!

**Comment by helto4real** [comment]
No it should not be Windows only, The containers we provide is Linux based.Had no problems compiling and running it on Linux. When you say "package" what do you mean ?

**Reply by ab5000** [reply]
Bear with me.  I'm learning about 5 things at once, so my language isn't precise, and probably inaccurate.

I'm pulling the NetDaemon packages via NuGet and building my own container.  I'm not using the containers you provided, like I said - I've gone a bit rogue, and may be in unsupported territory.  I made the post in the chance you find it helpful.  (e.g. it feels like the times when I've left debug / scaffolding code in by accident).  I'm not blocked, and I really appreciate the work you've put into this project!
 
When I "Run Dockerfile" in Rider with a hello world Main Program:
Console.WriteLine("Hello, World!");

with the following packages from NuGet, I get the expected output

`<ItemGroup>      
      <PackageReference Include="NetDaemon.AppModel" Version="24.18.0-alpha-4" />
      <PackageReference Include="NetDaemon.Client" Version="24.18.0-alpha-4" />
      <PackageReference Include="NetDaemon.Extensions.Scheduling" Version="24.18.0-alpha-4" />
      <PackageReference Include="NetDaemon.Extensions.Tts" Version="24.18.0-alpha-4" />
      <PackageReference Include="NetDaemon.Runtime" Version="24.18.0-alpha-4" />
    </ItemGroup>`

However, when I add (but change no code at all)
`<PackageReference Include="NetDaemon.Extensions.Logging" Version="24.18.0-alpha-4" />`

The Container no longer runs properly and I get the following runtime error.

> You must install or update .NET to run this application.
2024-05-18T19:32:21.414523980Z 
2024-05-18T19:32:21.414526424Z App: /app/bin/Debug/net8.0/HassConsolePlay2.dll
2024-05-18T19:32:21.414528087Z Architecture: x64
2024-05-18T19:32:21.414529700Z Framework: 'Microsoft.AspNetCore.App', version '8.0.0' (x64)
2024-05-18T19:32:21.414531273Z .NET location: /usr/share/dotnet/
2024-05-18T19:32:21.414532816Z 
2024-05-18T19:32:21.414534199Z No frameworks were found.
2024-05-18T19:32:21.414535612Z 
2024-05-18T19:32:21.414536994Z Learn more:
2024-05-18T19:32:21.414538417Z https://aka.ms/dotnet/app-launch-failed
2024-05-18T19:32:21.414539840Z 
2024-05-18T19:32:21.414541232Z To install missing framework, download:
2024-05-18T19:32:21.414542705Z https://aka.ms/dotnet-core-applaunch?framework=Microsoft.AspNetCore.App&framework_version=8.0.0&arch=x64&rid=linux-x64&os=debian.12

Removing the package library, fixes it.

**Reply by helto4real** [reply]
Hmm have no idea why. Are you using the official .net 8 images from Microsoft? You can check out the docker files in [the docker repo](https://github.com/net-daemon/docker/blob/main/base/Dockerfile8)
